test/studies/shootout/reverse-complement/bharshbarg/revcomp-mark-skip-read-begin.chpl

```diff
@@ -0,0 +1,87 @@
+/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+   contributed by Ben Harshbarger
+   derived from the Rust #2 version by Matt Brubeck
+*/
+
+const table = initTable("ATCGGCTAUAMKRYWWSSYRKMVBHDDHBVNN\n\n");
+
+config const readSize = 16 * 1024;
+
+proc main(args: [] string) {
+  const stdin = openfd(0);
+  var input = stdin.reader(iokind.native, locking=false);
+  var len = stdin.length();
+  var data : [0..#len] uint(8);
+  
+  sync { // wait for all process() tasks to complete before continuing
+
+    while true {
+      const descOffset = input._offset();
+      var nextDescOffset = descOffset;
+      var seqOffset = descOffset;
+      var eof = false;
+
+      // Mark where we start scanning (keep bytes in I/O buffer in input)
+      input._mark();
+
+      // Scan forward until we get to the \n (end of description)
+      input.advancePastByte(ascii("\n"));
+      seqOffset = input._offset();
+
+      try {
+        // Scan forward until we get to the > (end of sequence)
+        input.advancePastByte(ascii(">"));
+        nextDescOffset = input._offset();
+      } catch UnexpectedEOFError {
```

Comment:
Does the compiler currently support this pattern? My thought is that it would need to be `catch e: UnexpectedEOFError`.

Reply:
seems to work...

Reply:
Huh, cool.

Reply:
I bet that it's not matching on the type but rather I'm naming an error variable `UnexpectedEOFError` and it shadows the type. I.e. it's like I wrote `catch e`. I'll update it.